src/query/interpreter.cpp

```diff
@@ -1546,73 +1546,176 @@ std::vector<std::vector<TypedValue>> AnalyzeGraphQueryHandler::AnalyzeGraphCreat
+                                  });
+
+                    auto average_degree =
+                        no_vertices > 0 ? static_cast<double>(total_degree) / static_cast<double>(no_vertices) : 0;
+                    auto index_stats = storage::LabelIndexStats{.count = no_vertices, .avg_degree = average_degree};
+                    execution_db_accessor->SetIndexStats(label_id, index_stats);
+                    label_stats.emplace_back(std::make_pair(label_id, index_stats));
+                  });
+
+    return label_stats;
+  };
+
+  auto populate_label_property_stats = [&label_property_counter, &vertex_degree_counter, execution_db_accessor,
+                                        view](auto &index_info) {
+    // Iterate over all label property indexed vertices
+    std::for_each(
+        index_info.begin(), index_info.end(),
+        [execution_db_accessor, &label_property_counter, &vertex_degree_counter, view](const LPIndex &index_info) {
+          auto vertices = execution_db_accessor->Vertices(view, index_info.first, index_info.second);
+          std::for_each(vertices.begin(), vertices.end(),
+                        [&index_info, &label_property_counter, &vertex_degree_counter, &view](const auto &vertex) {
+                          label_property_counter[index_info][*vertex.GetProperty(view, index_info.second)]++;
+                          vertex_degree_counter[index_info] += *vertex.OutDegree(view) + *vertex.InDegree(view);
+                        });
         });
-    execution_db_accessor->SetIndexStats(
-        label_property.first, label_property.second,
-        storage::IndexStats{.statistic = chi_squared_stat, .avg_group_size = avg_group_size});
-    // Save result
-    result.emplace_back(execution_db_accessor->LabelToName(label_property.first));
-    result.emplace_back(execution_db_accessor->PropertyToName(label_property.second));
-    result.emplace_back(count_property_value);
-    result.emplace_back(static_cast<int64_t>(values_map.size()));
-    result.emplace_back(avg_group_size);
-    result.emplace_back(chi_squared_stat);
+
+    std::vector<std::pair<LPIndex, storage::LabelPropertyIndexStats>> label_property_stats;
+    label_property_stats.reserve(label_property_counter.size());
+    std::for_each(
+        label_property_counter.begin(), label_property_counter.end(),
+        [execution_db_accessor, &vertex_degree_counter, &label_property_stats](const auto &counter_entry) {
+          const auto &[label_property, values_map] = counter_entry;
+          // Extract info
+          uint64_t count_property_value = std::accumulate(
+              values_map.begin(), values_map.end(), 0,
+              [](uint64_t prev_value, const auto &prop_value_count) { return prev_value + prop_value_count.second; });
+          // num_distinc_values will never be 0
+          double avg_group_size = static_cast<double>(count_property_value) / static_cast<double>(values_map.size());
```

Comment:
```suggestion
          double avg_group_size = count_property_value / static_cast<double>(values_map.size());
```

Reply:
same above, Needed to cast this because of clang-tidy (narrowing conversion from uint64_t to double)